File: backend/routes/userRoute.js
import express from "express";
import authMiddleware from "../middleware/auth.js";
import { updateUserData,getUserData, loginUser, registerUser } from "../controllers/userController.js";
import multer from "multer";
import fs from "fs";
import jwt from "jsonwebtoken";
import userModel from "../models/userModel.js";

const userRouter = express.Router()

const storage = multer.diskStorage({
    destination: (req, file, cb) => {
      const token = req.headers.token;
      const decodedToken = jwt.verify(token, process.env.JWT_SECRET);
      const userId = decodedToken.id;
      userModel.findById(userId).then((user) => {
        const userFolder = `uploads/user_profiles/${user.customerId}`;
        fs.mkdirSync(userFolder, { recursive: true });
        cb(null, userFolder);
      });
    },
    filename: (req, file, cb) => {
      cb(null, file.originalname); // Store the file with the original name
    },
  });

const upload = multer({ storage: storage });

userRouter.post("/register",registerUser)
userRouter.post("/login",loginUser)
userRouter.get("/userdata",authMiddleware,getUserData)
userRouter.post("/updateuser",authMiddleware,upload.single("profileImg"),updateUserData)

export default userRouter;

File: food_app_rust_backend/src/routes/user_route.rs
use crate::controllers::user_controller::{
    get_user_data, login_user, register_user, update_user_data,
};
use crate::middleware::auth::auth_middleware;
use axum::extract::DefaultBodyLimit;

use axum::{
    routing::{get, post},
    Router,
};
// use tower_http::services::fs::FileSystem;
// use axum::extract::multipart::Multipart;

pub fn create_user_routes() -> Router {
    Router::new()
        .route("/api/user/register", post(register_user))
        .route("/api/user/login", post(login_user))
        .route(
            "/api/user/userdata",
            get(get_user_data).layer(axum::middleware::from_fn(auth_middleware)),
        )
        .route(
            "/api/user/updateuser",
            post(update_user_data).layer(DefaultBodyLimit::max(1024 * 100)).layer(axum::middleware::from_fn(auth_middleware)))  // 10 mb 
        
}
